src/gsi/setupgnssrspd.f90

```diff
@@ -0,0 +1,951 @@
+  !!awork(:) = zero
+
+  do i=1,nobs
+     muse(i)=nint(data(iuse,i)) <= jiter
+  end do
+
+  dup=one
+  do k=1,nobs
+     do l=k+1,nobs
+        if(data(ilat,k) == data(ilat,l) .and.  &
+           data(ilon,k) == data(ilon,l) .and.  &
+           data(ipres,k)== data(ipres,l) .and. &
+           data(ier,k) < r1000 .and. data(ier,l) < r1000 .and. &
+           muse(l) .and. muse(k))then
+
+           tfact=min(one,abs(data(itime,k)-data(itime,l))/dfact1)
+           dup(k)=dup(k)+one-tfact*tfact*(one-dfact)
+           dup(l)=dup(l)+one-tfact*tfact*(one-dfact)
+        end if
+     end do
+  end do
+
+  call dtime_setup()
+  do i=1,nobs
+     dtime=data(itime,i)
+     call dtime_check(dtime, in_curbin, in_anybin)
+     if(.not.in_anybin) cycle
+
+     if(in_curbin) then
+        dlat=data(ilat,i)
+        dlon=data(ilon,i)
+ 
+        dpres=data(ipres,i)
+        error=data(ier2,i)
+        ikx=nint(data(ikxx,i))
+     endif
+
+!    Link observation to appropriate observation bin
+     if (nobs_bins>1) then
+        ibin = NINT( dtime/hr_obsbin ) + 1
+     else
+        ibin = 1
+     endif
+     IF (ibin<1.OR.ibin>nobs_bins) write(6,*)mype,'Error nobs_bins,ibin= ',nobs_bins,ibin
+
+     if (luse_obsdiag) my_diagLL => odiagLL(ibin)
+
+!    Link obs to diagnostics structure
+     if (luse_obsdiag) then
+        my_diag => obsdiagLList_nextNode(my_diagLL      ,&
+                create = .not.lobsdiag_allocated        ,&
+                   idv = is             ,&
+                   iob = ioid(i)        ,&
+                   ich = 1              ,&
+                  elat = data(ilate,i)  ,&
+                  elon = data(ilone,i)  ,&
+                  luse = luse(i)        ,&
+                 miter = miter          )
+
+        if(.not.associated(my_diag)) call die(myname, &
+                'obsdiagLList_nextNode(), create =', .not.lobsdiag_allocated)
+     endif
+
+     if(.not.in_curbin) cycle
+
+!    Load obs error and u,v obs
+     obserror = max(cermin(ikx),min(cermax(ikx),data(ier,i)))
+     uob = data(iuob,i)
+     vob = data(ivob,i)
+ 
+ 
+     gnssrspdob=sqrt(uob*uob+vob*vob)
+     call tintrp2a1(ges_tv,tges,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+     call tintrp2a11(ges_ps,psges,dlat,dlon,dtime,hrdifsig,&
+          mype,nfldsig)
+     call tintrp2a1(ges_lnprsl,prsltmp,dlat,dlon,dtime,hrdifsig,&
+          nsig,mype,nfldsig)
+
+     factw = data(iff10,i)
+     if(sfcmod_gfs .or. sfcmod_mm5)then
+        sfcr = data(isfcr,i)
+        skint = data(iskint,i)
+        isli=data(idomsfc,i)
+        call comp_fact10(dlat,dlon,dtime,skint,sfcr,isli,mype,factw)
+     end if
+
+     nty=ictype(ikx)
+
+!    Process observations reported with height differently than those
+!    reported with pressure.  Type 260=nacelle 261=tower wind spd are
+!    encoded in NCEP prepbufr files with geometric height above
+!    sea level. 
+
+!    Process cygnss observations at mslp
+     if ( nty == 298) then
+          z_height = .true.
```

Comment:
`z_height` is not defined if `nty /= 298`.  It is dangerous to assume that unfinalized logical variables default to `.false.`   Provide a default value for `z_height`

Reply:
I initialized z_height to false. Good catch!